### PR TITLE
Port `issue-1305.sli` to Pytest

### PR DESCRIPTION
**Pull request by nicolossus**

### CODE REVIEW

File: testsuite/pytests/sli2py_regressions/test_issue_1305.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Regression test for Issue #1305 (GitHub).
+
+This set of tests ensures that NEST hcan set small resolutions and deals with
```
Comment (janskaar) [inline testsuite/pytests/sli2py_regressions/test_issue_1305.py:25]:
```suggestion
This set of tests ensures that NEST can set small resolutions and deals with
```

**Review by janskaar — APPROVED** [review]

File: testsuite/pytests/sli2py_regressions/test_issue_1305.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def test_resolution_rounding_valid():
+    """Test setting valid resolution."""
+
+    target_resolution = 0.102
+    nest.set(resolution=target_resolution, tics_per_ms=1000.0)
+
+    abs_diff = abs(nest.resolution - target_resolution)
+    assert abs_diff < 1e-15
```
Comment (heplesser) [inline testsuite/pytests/sli2py_regressions/test_issue_1305.py:46]: Please use `pytest.approx` here.

**Review by heplesser — CHANGES_REQUESTED** [review]
See inline. The precise value of 1e-15 in the test is not important, a relative error of 1e-6 as default for approx is fine.

File: testsuite/pytests/sli2py_regressions/test_issue_1305.py (hunk 2 above)
Comment (nicolossus) [reply testsuite/pytests/sli2py_regressions/test_issue_1305.py]: Fixed by https://github.com/nest/nest-simulator/pull/2806/commits/32f111fcd83991a2297ca3323519bf1e77a7873b.

**Review by heplesser — APPROVED** [review]

**Pull request merged**